environment/filesystem.go

```diff
@@ -60,3 +78,49 @@ func (env *Environment) FileList(ctx context.Context, path string) (string, erro
 	}
 	return out.String(), nil
 }
+
+func (env *Environment) FileGrep(ctx context.Context, path, pattern, include string) (string, error) {
+	// Hack: use busybox to run `sed` since dagger doesn't have native file editing primitives.
+	args := []string{"/bin/grep", "-E", "--", pattern, include}
```

Comment:
What about using `ripgrep` instead? It's smart enough to avoid grepping through `node_modules` and other crap